src/containers/Tenant/Query/QueryResult/components/QueryInfoDropdown/useQueryInfoMenuItems.tsx

```diff
@@ -0,0 +1,179 @@
+            const handleGetSvg = () => {
+                if (blobUrl) {
+                    return Promise.resolve(blobUrl);
+                }
+
+                return getPlanToSvg({plan, database})
+                    .unwrap()
+                    .then((result) => {
+                        const blob = new Blob([result], {type: 'image/svg+xml'});
+                        const url = URL.createObjectURL(blob);
+                        setBlobUrl(url);
+                        return url;
+                    })
+                    .catch((err) => {
+                        const errorMessage = prepareCommonErrorMessage(err);
+                        createToast({
+                            title: i18n('text_error-plan-svg', {error: errorMessage}),
+                            name: 'plan-svg-error',
+                            type: 'error',
+                        });
+                        return null;
+                    });
+            };
+
+            const handleOpenInNewTab = () => {
+                handleGetSvg().then((url) => {
+                    if (url) {
+                        window.open(url, '_blank');
+                    }
+                });
+            };
+
+            const handleDownload = () => {
+                handleGetSvg().then((url) => {
+                    const link = document.createElement('a');
+                    if (url) {
+                        link.href = url;
+                        link.download = 'query-plan.svg';
+                        document.body.appendChild(link);
+                        link.click();
+                        document.body.removeChild(link);
+                    }
+                });
+            };
+
+            menuItems.push([
+                {
+                    text: (
+                        <MenuItemContent
+                            title={i18n('text_open-new-tab')}
+                            description={i18n('text_open-new-tab_description')}
+                        />
+                    ),
+                    icon: <ArrowUpRightFromSquare className={b('icon')} />,
+                    action: handleOpenInNewTab,
+                    className: b('menu-item'),
+                },
+                {
+                    text: (
+                        <MenuItemContent
+                            title={i18n('text_download')}
+                            description={i18n('text_download_description')}
+                        />
+                    ),
+                    icon: <ArrowDownToLine className={b('icon')} />,
+                    action: handleDownload,
+                    className: b('menu-item'),
+                },
+            ]);
+        }
+
+        if (queryResultsInfo) {
+            const handleDiagnosticsDownload = () => {
+                const diagnosticsData = {
+                    ...queryResultsInfo,
+                    database,
+                    timestamp: new Date().toISOString(),
```

Comment:
If I were user, I would expect that in `timestamp` field it is some query stats - start or finish. What do you think about renaming this property?

Reply:
removed